moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1736,7 +1714,12 @@ bool PlanningScene::processCollisionObjectAdd(const moveit_msgs::CollisionObject
   if (object.operation == moveit_msgs::CollisionObject::ADD && world_->hasObject(object.id))
     world_->removeObject(object.id);
 
-  const Eigen::Isometry3d& object_frame_transform = getTransforms().getTransform(object.header.frame_id);
+  const Eigen::Isometry3d& world_to_object_header_transform = getTransforms().getTransform(object.header.frame_id);
+  Eigen::Isometry3d object_header_to_pose_transform;
+                                                    
+  PlanningScene::poseMsgToEigen(object.pose, object_header_to_pose_transform);
+
+  const Eigen::Isometry3d& object_frame_transform = world_to_object_header_transform * object_header_to_pose_transform;
```

Comment:
Same comment as above on this one, suggest dropping the reference here.

Reply:
Done